### PR TITLE
CI: run misspell check, gofmt check and staticcheck

### LINKED ISSUE #86: code quality checks in CI
it looks like our CI doesn't enforce any code quality checks.
what comes closest is the readme linking to https://goreportcard.com/report/github.com/naggie/dstask
in another project I work on, I have a rich set of [qa checks that enforced](https://github.com/grafana/metrictank/tree/master/scripts/qa) (note, a couple of them are specific to the project) and also some [subjective ones that are not enforced](https://github.com/grafana/metrictank/tree/master/scripts/qa-subjective)

i mention this because there's some interesting examples there to check for spelling mistakes, health of vendor directory, whether everything is correctly formatted, finding unused code, etc.

These scripts may be a bit outdated now though. This space is pretty crowded. https://staticcheck.io/ seems quite solid, there's also gometalinter, go vet (which has both enforceable and more subjective tests), etc.
I'm not sure what's the current best practice. I'm hopeful the community has converged around "one tool to rule them all" ?

### PR DESCRIPTION
**Pull request by Dieterbe**
fix #86

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -343,7 +343,7 @@ func CommandOpen(conf Config, ctx, cmdLine CmdLine) error {
 	for _, task := range ts.Tasks() {
 		urls := xurls.Relaxed.FindAllString(task.Summary+" "+task.Notes, -1)
 		if len(urls) == 0 {
-			return fmt.Errorf("No URLs found in task %v", task.ID)
+			return fmt.Errorf("no URLs found in task %v", task.ID)
```
Comment (naggie) [inline commands.go:346]: Any reason for un-capitalising? Seems less neat

**Comment by naggie** [comment]
Thanks, very useful

File: commands.go (hunk 1 above)
Comment (Dieterbe) [reply commands.go:346]: not sure if this is an official best practice or merely a best practice recommended by many code quality tools, but at least the official examples don't capitalize errors , e.g. see https://blog.golang.org/error-handling-and-go

Comment (naggie) [reply commands.go:346]: Never heard of it but never looked. I'm ok with it though and the docs are good enough of an indication of best practice 👍

**Pull request merged**